cluster-sync/sync.sh

```diff
@@ -79,6 +80,40 @@ function wait_cdi_available {
   fi
 }
 
+OLD_CDI_VER_PODS="./_out/tests/old_cdi_ver_pods"
+NEW_CDI_VER_PODS="./_out/tests/new_cdi_ver_pods"
+
+# Wait for update of all cdi pods (names, uids and image versions)
+# Note it will fail update to the same version
+function wait_cdi_pods_updated {
+  echo "Waiting $CDI_PODS_UPDATE_TIMEOUT seconds for all CDI pods to update"
+  if [ -f $NEW_CDI_VER_PODS ] ; then
+    mv $NEW_CDI_VER_PODS $OLD_CDI_VER_PODS
+  fi
+  wait_time=0
+  ret=0
+  while [[ $ret -eq 0 ]] && [[ $wait_time -lt ${CDI_PODS_UPDATE_TIMEOUT} ]]; do
+    wait_time=$((wait_time + 5))
+    _kubectl get pods -n $CDI_NAMESPACE -o=jsonpath='{range .items[*]}{.metadata.name}{"\n"}{.metadata.uid}{"\n"}{.spec.containers[*].image}{"\n"}{end}' > $NEW_CDI_VER_PODS
+    if [ -f $OLD_CDI_VER_PODS ] ; then
+      grep -qFxf $OLD_CDI_VER_PODS $NEW_CDI_VER_PODS || ret=$?
```

Comment:
Will this return non zero when there is any difference between old and new?  If so, is that really sufficient to say that all pods have changed?

Reply:
@mhenriks no, grep will return non-zero when **none** of the lines (pod name, uid or image) on the old file appears in the new file, which sure means all pods have changed. However it does not check all the new pods are Running etc., which out of scope here.